### PR TITLE
Array shift

### PR DESCRIPTION
**Pull request by F0rgiv**

### CODE REVIEW

**Comment by F0rgiv** [comment]
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] expected use case tests are passing
     - [x] Edge Cases implemented such as empty array
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Picture of whiteboard

File: challenges/lib/src/main/java/challenges/Library.java [hunk 1]
```diff
@@ -4,7 +4,31 @@
 package challenges;
 
 public class Library {
-    public boolean someLibraryMethod() {
-        return true;
+    public static int binarySearch(int[] arr, int target){
+      //if empty return -1
```
Comment (Mattpet26) [inline challenges/lib/src/main/java/challenges/Library.java:8]: Nice, always check for the base case. No point in doing all this work if our given input was empty.

File: challenges/lib/src/test/java/challenges/LibraryTest.java [hunk 2]
```diff
@@ -7,8 +7,21 @@
 import static org.junit.Assert.*;
 
 public class LibraryTest {
-    @Test public void testSomeLibraryMethod() {
-        Library classUnderTest = new Library();
-        assertTrue("someLibraryMethod should return 'true'", classUnderTest.someLibraryMethod());
-    }
+ @Test
```
Comment (Mattpet26) [inline challenges/lib/src/test/java/challenges/LibraryTest.java:10]: Great work on the tests. Multiple tests + checking what happens if the array is empty!
